gtk4/src/subclass/cell_layout.rs

```diff
@@ -0,0 +1,368 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::{CellArea, CellLayout, CellRenderer};
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::Cast;
+
+#[derive(Debug)]
+pub struct CellLayoutDataCallback {
+    callback: ffi::GtkCellLayoutDataFunc,
+    user_data: glib::ffi::gpointer,
+    destroy_notify: glib::ffi::GDestroyNotify,
```

Comment:
Same as elsewhere: memory management and making it callable

Reply:
Not sure how I'm supposed to figure out the memory management here? I suppose the difference between this callback & the others on this merge request is the destroy_notify. Does the when to call the `destroy_notify` is the figuring memory management part?

Reply:
You need to call it as part of the `Drop` impl on this type

Reply:
Ah, I see. Thanks!

Reply:
And for calling you'd add some kind of `rn call(&self, all, the arguments) -> return_type` that internally calls `callback` with `user_data` and the other arguments.

Reply:
Yeah looks good how you implemented that now. You'd then pass a `CellLayoutDataCallback` through all the Rust code so that implementors can call it whenever they want and also store it away somewhere for calling it later.